packages/flutter/lib/src/material/progress_indicator_theme.dart

```diff
@@ -0,0 +1,170 @@
+  ///    a an animated color.
+  final Color? color;
+
+  /// Creates a copy of this object but with the given fields replaced with the
+  /// new values.
+  ProgressIndicatorThemeData copyWith({
+    Color? backgroundColor,
+    Color? color,
+  }) {
+    return ProgressIndicatorThemeData(
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      color: color ?? this.color,
+    );
+  }
+
+  /// Linearly interpolate between two progress indicator themes.
+  ///
+  /// If both arguments are null, then null is returned.
+  static ProgressIndicatorThemeData? lerp(ProgressIndicatorThemeData? a, ProgressIndicatorThemeData? b, double t) {
+    if (a == null && b == null)
+      return null;
+    assert(t != null);
+    return ProgressIndicatorThemeData(
+      backgroundColor: Color.lerp(a?.backgroundColor, b?.backgroundColor, t),
+      color: Color.lerp(a?.color, b?.color, t),
+    );
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      backgroundColor,
+      color,
+    );
+  }
+
+  @override
+  bool operator==(Object other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is ProgressIndicatorThemeData
+        && other.backgroundColor == backgroundColor
+        && other.color == color;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(ColorProperty('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(ColorProperty('color', color, defaultValue: null));
+  }
+}
+
+/// An inherited widget that defines the configuration for
+/// [ProgressIndicator]s in this widget's subtree.
+///
+/// Values specified here are used for [ProgressIndicator] properties that are not
+/// given an explicit non-null value.
+///
+/// {@tool snippet}
+///
+/// Here is an example of a progress indicator theme that applies a red indicator
+/// color with a slightly transparent red for the background track color.
+///
+/// ```dart
+/// ProgressIndicatorTheme(
+///   data: ProgressIndicatorThemeData(
+///     color: Colors.red,
+///     backgroundColor: Colors.red.withOpacity(0.25),
+///   ),
+///   child: LinearProgressIndicator()
+/// ),
+/// ```
+/// {@end-tool}
+class ProgressIndicatorTheme extends InheritedTheme {
+  /// Creates a theme that controls the configurations for [ProgressIndicator]
+  /// widgets.
+  const ProgressIndicatorTheme({
+    Key? key,
+    required this.data,
+    required Widget child,
+  }) : assert(data != null), super(key: key, child: child);
+
+  /// The properties for descendant [ProgressIndicator] widgets.
+  final ProgressIndicatorThemeData data;
+
+  /// Returns the [data] from the closest [ProgressIndicatorTheme] ancestor. If
+  /// there is no ancestor, it returns [ThemeData.progressIndicatorTheme].
+  /// Applications can assume that the returned value will not be null.
```

Comment:
We don't need to say this (anymore) because the return type is not `ProgressIndicatorThemeData?`